swift/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022-2023 Chartboost, Inc.
+// 
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+//
+//  FullscreenAdController.swift
+//  ChartboostMediationDemo
+//
+//  Copyright © 2023 Chartboost. All rights reserved.
+//
+
+import UIKit
+import ChartboostMediationSDK
+
+/// A basic implementation of a controller for Chartboost Mediation fullscreen ads.  It is capable of loading and showing a full screen fullscreen ad
+/// for a single placement.  This controller is also its own `CHBHeliumFullscreenAdDelegate` so that it is in full control
+/// of the ad's lifecycle.
+class FullscreenAdController: NSObject {
+    /// The entry point for the Chartboost Mediation SDK.
+    private let chartboostMediation = Helium.shared()
+
+    /// Which type of fullscreen ad is managed by this controller
+    private let adType: AdType?
+
+    /// The placement that is controller is for.
+    private var placementName: String
+
+    /// An instance of the fullscreen ad that this class controls the lifecycle of.
+    private var fullscreenAd: ChartboostMediationFullscreenAd?
+
+    /// A delegate for demo purposes only so that long activity processes can be communicated to a view controller.
+    private weak var activityDelegate: ActivityDelegate?
+
+    /// Initializer for the view model.
+    /// - Parameter placementName: Placement to use.
+    /// - Parameter activityDelegate: A delegate to communicate the start and end of asyncronous activity to.  This is applicable only for this demo.
+    init(adType: AdType?, activityDelegate: ActivityDelegate) {
+        self.adType = adType
+        self.activityDelegate = activityDelegate
+
+        switch adType {
+        case .interstitial:
+            placementName = "AllNetworkInterstitial"
+        case .rewarded:
+            placementName = "AllNetworkRewarded"
+        // If adType is nill or not a fullscreen type, give an empty placement name that won't load
+        default:
+            placementName = ""
+        }
+    }
+
+    /// Load the fullscreen ad.
+    /// - Parameter keywords: Optional keywords that can be associated with the advertisement placement.
+    func load(keywords: HeliumKeywords? = nil) {
+        // Attempt to load the ad only if it has not already been created and requested to load.
+        guard fullscreenAd == nil else {
+            print("[Warning] fullscreen advertisement has already been loaded")
+            return
+        }
+
+        // Notify the demo UI
+        activityDelegate?.activityDidStart()
+
+        // loadFullscreenAd expects keywords to be `[String : String]` instead of `HeliumKeywords?`.
+        let keywords = keywords?.dictionary ?? [:]
```

Comment:
Instead of having keywords = nil passed in, just build an empty dictionary to pass and leave a comment about how this is where you'd add any keywords